corehq/apps/hqadmin/management/commands/stale_data_in_es.py

```diff
@@ -175,43 +201,65 @@ def get_sql_chunks(run_config):
             yield matching_records
 
     @staticmethod
-    def _yield_missing_in_es(chunk):
+    def _yield_missing_in_es(chunk, dedicated_case_search_index=None):
```

Comment:
The existing version of this operates on a chunk that might contain cases that should be present in the `case_search` index and some that shouldn't be (that's what `_get_ids_from_chunk` does).  I don't think that logic is relevant anymore, if all domains are on the case search index.  Still, now we have two separate ways of splitting the logic for two separate per-domain settings, which is kind of weird.  Have you looked at simplifications or doing the same logic for both?

Reply:
are you referring to `if domain_needs_search_index(domain):`? I was trying to figure out if I can get rid of it but I have not gotten a response on [slack](https://dimagi.slack.com/archives/C0B1Z40DA/p1717765781969359) yet